central/complianceoperator/v2/report/manager/generator/report_gen_impl.go

```diff
@@ -101,7 +101,10 @@ func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *report.Reques
 	if snapshot != nil {
 		snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_GENERATED
 		if len(req.FailedClusters) > 0 {
-			snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_PARTIAL_ERROR
+			snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_PARTIAL_SCAN_ERROR_EMAIL
+			if req.NotificationMethod == storage.ComplianceOperatorReportStatus_DOWNLOAD {
+				snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_PARTIAL_SCAN_ERROR_DOWNLOAD
+			}
```

Comment:
That is fully okay, as we have 2 options only. My first thought would be to use a switch here

```go
switch(req.NotificationMethod) {
    case storage.ComplianceOperatorReportStatus_DOWNLOAD:
        // foo
    case storage.ComplianceOperatorReportStatus_EMAIL:
        // bar
}
```